### PR TITLE
Added a yaml store

### PR DESCRIPTION
**Pull request by hamnis**
Added a yaml store.

This is basically a copy of the json store, just instead using yaml.

This pull request is just a way to start a conversation about how we can use yaml as a store.
It is for instance possible to have multiple document in a yaml store, meaning it can work as 
properties store, and it can have tags, so we could in theory support tagging within the store itself.
For now this is just a stupid serialization back to json, so we have basically exactly the same functionality as the  json store. But with comments ;)

### CODE REVIEW

**Comment by zapodot** [comment]
Will have a look at it ASAP

File: constretto-core/src/test/java/org/constretto/internal/store/YamlStoreTest.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                .addResource(Resource.create("file:/nonExisiting.yaml"), "person")
+                .addResource(Resource.create("http:/nonExisiting.com"), "person")
+                .done()
+                .getConfiguration();
+
+        assertFalse(conf.hasValue("person"));
+    }
+
+
+    private class PersonJsonConverter implements GenericConverter<Person> {
```
Comment (zapodot) [inline constretto-core/src/test/java/org/constretto/internal/store/YamlStoreTest.java]: Json? ;-)

File: constretto-core/src/test/java/org/constretto/internal/store/YamlStoreTest.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                .createYamlConfigurationStore()
+                .addResource(Resource.create("classpath:yamlTest2.yaml"), "person")
+                .done()
+                .getConfiguration();
+        Person person = conf.evaluateWith(new PersonJsonConverter(), "person");
+        assertEquals(new Person("Erlend", 34), person);
+    }
+
+    @Test
+    public void jsonShouldAllowNonExistingResources() {
```
Comment (zapodot) [inline constretto-core/src/test/java/org/constretto/internal/store/YamlStoreTest.java]: I guess you really mean yaml...

File: constretto-core/src/test/java/org/constretto/internal/store/YamlStoreTest.java (hunk 1 above)
Comment (hamnis) [inline constretto-core/src/test/java/org/constretto/internal/store/YamlStoreTest.java]: well not really. this is converted to json, so at this stage it is json.

File: constretto-core/src/test/java/org/constretto/internal/store/YamlStoreTest.java (hunk 2 above)
Comment (hamnis) [inline constretto-core/src/test/java/org/constretto/internal/store/YamlStoreTest.java]: yeah

File: constretto-core/src/test/java/org/constretto/internal/store/YamlStoreTest.java (hunk 1 above)
Comment (zapodot) [inline constretto-core/src/test/java/org/constretto/internal/store/YamlStoreTest.java]: :-)

**Comment by zapodot** [comment]
In the Constretto-Dropwizard lib there we have used an approach that enables us to use tagging within yaml files:
```yaml

database:
  # the JDBC URL
  url: jdbc:oracle:thin:@//oracle-testing:1521/name
  .staging.url: jdbc:oracle:thin:@//oracle-staging:1521/name
  .production.url: jdbc:oracle:thin:@//oracle-production:1521/name

```

Using "." instead of "@" keeps the yaml code valid. It would be great to see that integrated to Constretto core as well! What do you think @hamnis .., Would that be useful for you to?

**Pull request merged**